### PR TITLE
Fix stomp mode crash

### LINKED ISSUE #118: Stomp mode game crash
I noticed the game would sometimes crash during stomp mode, and I was able to create a tour that crashes the game 100% of the time.
[CrashTour.zip](https://github.com/mmatyas/supermariowar/files/3829034/CrashTour.zip)

### PR DESCRIPTION
**Pull request by AzureBerylBlue**
Fixes #118

### CODE REVIEW

File: src/common/eyecandy.cpp [hunk 1]
```diff
@@ -409,17 +409,19 @@ void EC_Corpse::update()
...
-            IO_Block * leftblock = g_map->block(tx, ty);
-            IO_Block * rightblock = g_map->block(tx2, ty);
-
-            if ((g_map->map(tx, ty) & 0x13) > 0 || (g_map->map(tx2, ty) & 0x13) > 0 ||
-                    (leftblock && !leftblock->isTransparent() && !leftblock->isHidden()) || (rightblock && !rightblock->isTransparent() && !rightblock->isHidden())) {
-                //on ground
-                dy = (float) ((ty << 5)  - TILESIZE);
-                vely = 0.0f;
-                iy = (short)dy;
+            if (tx >= 0 && tx2 >= 0 && tx < smw->ScreenWidth / TILESIZE &&  tx2 < smw->ScreenWidth / TILESIZE) {
```
Comment (mmatyas) [inline src/common/eyecandy.cpp]: There's a duplicate space :)

```
TILESIZE &&  tx2 <
           ^
```

**Comment by mmatyas** [comment]
It does seem to fix the issue, but a few things came to my mind while reviewing:

1. Would it be possible for `ty` to be invalid too?
2. Why do `tx`/`tx2` have incorrect values in the first place? Perhaps we could prevent the issue there too (they seem to be set in the `EC_Corpse` constructor).
3. How many other parts of the code could be vulnerable to similar out of range index bugs? I see this `pointer = g_map->block(...)` then `if (pointer && ...)` pattern several places, I wonder if adding a range check to `g_map->block()` instead would make it more error resistant.

**Comment by AzureBerylBlue** [comment]
Ok, I updated it. I think the problem was that it was not wrapping the coordinates across the screen correctly.

**Comment by mmatyas** [comment]
Yup confirmed working :+1:

**Pull request merged**